src/Bicep.Core/Emit/PositionTrackingJsonTextWriter.cs

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using Newtonsoft.Json;
+
+namespace Bicep.Core.Emit
+{
+    public class PositionTrackingJsonTextWriter : JsonTextWriter
+    {
+        private class PositionTrackingTextWriter : TextWriter
+        {
+            private readonly TextWriter _internalWriter;
```

Comment:
[nit] The convention for the Bicep repo is to not prefix field names with `_` and use `this.` to reference them.